crates/router/src/connector/trustpay/transformers.rs

```diff
@@ -849,23 +849,33 @@ impl TryFrom<&types::PaymentsPreProcessingRouterData> for TrustpayCreateIntentRe
             .as_ref()
             .map(|pmt| matches!(pmt, diesel_models::enums::PaymentMethodType::GooglePay));
 
+        let request_amount = item
```

Comment:
In Preprocessing amount and currency need not to be optional fields. Please check this if any changes required take it in separate PR